quickcheck/src/main/java/fj/test/Arbitrary.java

```diff
@@ -1337,4 +1338,10 @@
     return aa.bind(ab, ac, ad, ae, af, ag, ah,
         a -> b -> c -> d -> e -> f -> g -> h -> p(a, b, c, d, e, f, g, h));
   }
+
+  /**
+   * An arbitrary implementation for naturals.
+   */
+  public static Gen<Natural> arbNatural = arbLong.map(Longs.abs).map(Natural::natural).map(o -> o.some());
```

Comment:
I think it would be nice if the generator could generate some values above ```Long#MAX_VALUE```. But I'm not sure what is the best strategy for doing that... WDYT?